packages/yarnpkg-core/sources/Installer.ts

```diff
@@ -21,7 +21,30 @@ export type FinalizeInstallStatus = {
   buildDirective: Array<BuildDirective>,
 };
 
+export type PackageMeta = Record<any, any>;
+
 export interface Installer {
+  /**
+   * Returns cache key used to store package meta information.
+   *
+   * The key should be unique to the linker, the package and the meta format version used.
+   *
+   * @param pkg The package being installed
+   */
+  getPackageMetaKey?(pkg: Package): string;
+
+  /**
+   * Retrieves serializable meta information from the fetched information about the package.
```

Comment:
Same, the comment should use the PnP linker to explain why this function matters for us and what are the information we typically store inside the meta.